api/tokens.go

```diff
@@ -148,3 +150,171 @@ func (api *API) RefreshHandler(ctx context.Context, w http.ResponseWriter, req *
 
 	return models.NewSuccessResponse(jsonResponse, http.StatusCreated, headers), nil
 }
+
+//SignOutAllUsersHandler bulk refresh token invalidation for panic sign out handling
+func (api *API) SignOutAllUsersHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	var (
+		userFilterString string = "status=\"Enabled\""
+		backOff = []time.Duration{
+			1 * time.Second,
+			3 * time.Second,
+			10 * time.Second,
+		}
+	)
+	usersList, awsErr := api.ListUsersWorker(req.Context(), &userFilterString, backOff)
+	if awsErr != nil {
+		return nil, awsErr
+	}
+	globalSignOut := &models.GlobalSignOut{
+		ResultsChannel: make(chan string, len(*usersList)),
+		BackoffSchedule: backOff,
+		RetryAllowed: true,
+	}
+	// run api.SignOutUsersWorker concurrently
+	go api.SignOutUsersWorker(req.Context(), globalSignOut, usersList)
+	return models.NewSuccessResponse(nil, http.StatusAccepted, nil), nil
+}
+
+// ListUsersWorker - generates a list of users based on `userFilterString` filter string
+func (api *API) ListUsersWorker(ctx context.Context, userFilterString *string, backoffSchedule []time.Duration) (*[]models.UserParams, *models.ErrorResponse) {
+	var (
+		awsErr error
+		paginationToken *string
+		usersList models.UsersList
+		listUsersResp, result *cognitoidentityprovider.ListUsersOutput
+		listUserInput = usersList.BuildListUserRequest(
+			*userFilterString,
+			"",
+			int64(0), 
+			nil,
+			&api.UserPoolId,
+		)
+		usersListError *models.ErrorResponse
+	)
+	listUsersResp, awsErr = api.generateListUsersRequest(listUserInput)
```

Comment:
I think this could be simplified to something like this
```suggestion
	loadingInProgress := true
	for loadingInProgress {
		for _, backoff := range backoffSchedule {
			result, awsErr = api.generateListUsersRequest(listUserInput)
			if awsErr == nil {
				listUsersResp.Users = append(listUsersResp.Users, result.Users...)
				if result.PaginationToken != nil {
					listUserInput.PaginationToken = result.PaginationToken
				} else {
					loadingInProgress = false
				}
			} else {
				err := models.NewCognitoError(ctx, awsErr, "Cognito ListUsers request from signout all users from group endpoint")
				if err.Code != models.TooManyRequestsError {
					return nil, models.NewErrorResponse(http.StatusInternalServerError, nil, err)
				}
				time.Sleep(backoff)
			}
		}
	}
	usersList.MapCognitoUsers(listUsersResp)
	return &usersList.Users, nil
```